examples/medplum-mso-demo/src/utils/enrollment.ts

```diff
@@ -0,0 +1,297 @@
+  Patient,
+  Practitioner,
+  Reference,
+  ProjectMembership,
+  AccessPolicy,
+  ProjectMembershipAccessParameter,
+} from '@medplum/fhirtypes';
+import { createReference, MedplumClient, normalizeErrorString, getReferenceString, resolveId } from '@medplum/core';
+
+/**
+ * Enrolls a practitioner in an organization by adding the organization to the access array of the practitioner's project membership.
+ *
+ *  1. Search for the practitioner's project membership, if there is no project membership, throw an error
+ *  2. Check if the organization reference already exists in any access array
+ *  3. If the organization reference does not exist, add the organization to the ProjectMembership.access array
+ *  4. Update the ProjectMembership resource
+ *
+ * @param medplum - The Medplum client.
+ * @param practitioner - The practitioner to enroll.
+ * @param organization - The organization to enroll the practitioner in.
+ * @returns The updated project membership resource
+ */
+export async function enrollPractitioner(
+  medplum: MedplumClient,
+  practitioner: Practitioner,
+  organization: Organization
+): Promise<ProjectMembership> {
+  // 1. Search for the practitioner's project membership, if there is no project membership, throw an error
+  const projectMembershipSearch = await medplum.searchOne('ProjectMembership', {
+    profile: getReferenceString(practitioner),
+  });
+
+  if (projectMembershipSearch) {
+    const membershipResource = projectMembershipSearch;
+    const existingAccess = membershipResource.access || [];
+
+    // 2. Check if this organization reference already exists in any access array
+    const organizationExists = existingAccess.some((access) =>
+      access.parameter?.some(
+        (param) => param.name === 'organization' && param.valueReference?.reference === getReferenceString(organization)
+      )
+    );
+
+    // 3. If the organization reference does not exist, add the organization to the ProjectMembership.access array
+    if (!organizationExists) {
+      const policy = await getAccessPolicyByName(medplum, 'Managed Service Organization Access Policy');
+
+      if (existingAccess.length > 0) {
+        existingAccess.push({
+          parameter: [{ name: 'organization', valueReference: createReference(organization) }],
+          policy: { reference: getReferenceString(policy) },
+        });
+      } else {
+        membershipResource.access = [
+          {
+            parameter: [{ name: 'organization', valueReference: createReference(organization) }],
+            policy: { reference: getReferenceString(policy) },
+          },
+        ];
+      }
+    }
+
+    // 4. Update the ProjectMembership resource
+    try {
+      const updatedResource = await medplum.updateResource(membershipResource);
+      return updatedResource;
+    } catch (error) {
+      throw new Error(normalizeErrorString(error));
+    }
+  }
+  throw new Error(`No project membership found for practitioner ${practitioner.id}`);
```

Comment:
Out of scope for this PR, but a future version of this should demonstrate how to invite Users as well. The who project vs. server scoped users is a common source of confusion, and we should have a best practice recommendation here